src/ray/core_worker/reference_count.cc

```diff
@@ -770,14 +776,9 @@ int64_t ReferenceCounter::EvictLineage(int64_t min_bytes_to_evict) {
 }
 
 void ReferenceCounter::ReleasePlasmaObject(ReferenceTable::iterator it) {
-  if (it->second.on_delete) {
-    RAY_LOG(DEBUG) << "Calling on_delete for object " << it->first;
-    it->second.on_delete(it->first);
-    it->second.on_delete = nullptr;
-  }
```

Comment:
`ReleasePlasmaObject` can be called even when the object is still in scope (e.g. the primary copy is lost) and in this case, we shouldn't call the `on_delete` (aka `on_out_of_scope`) callback.